nano/nano_node/entry.cpp

```diff
@@ -402,10 +403,30 @@ int main (int argc, char * const * argv)
 				for (auto i (0); i != num_accounts; ++i)
 				{
 					genesis_balance = genesis_balance - 1000000000;
-					auto send (std::make_shared<nano::state_block> (nano::test_genesis_key.pub, genesis_latest, nano::test_genesis_key.pub, genesis_balance, keys[i].pub, nano::test_genesis_key.prv, nano::test_genesis_key.pub, work.generate (genesis_latest)));
+
+					auto send = builder.state ()
+					            .account (nano::test_genesis_key.pub)
+					            .previous (genesis_latest)
+					            .representative (nano::test_genesis_key.pub)
+					            .balance (genesis_balance)
+					            .link (keys[i].pub)
+					            .sign (nano::test_genesis_key.prv, nano::test_genesis_key.pub)
+					            .work (work.generate (genesis_latest))
+					            .build ();
+
 					genesis_latest = send->hash ();
 					blocks.push_back (std::move (send));
-					auto open (std::make_shared<nano::state_block> (keys[i].pub, 0, keys[i].pub, balances[i], genesis_latest, keys[i].prv, keys[i].pub, work.generate (keys[i].pub)));
+
+					auto open = builder.state ()
+					            .account (keys[i].pub)
+					            .previous (0)
+					            .representative (keys[i].pub)
+					            .balance (balances[i])
+					            .link (genesis_latest)
+					            .sign (nano::test_genesis_key.prv, nano::test_genesis_key.pub)
```

Comment:
Should signing use the same keys as was used before? `keys[i].prv, keys[i].pub`

Reply:
Yeah it should, thanks!